lib/datadog/tracing/distributed/propagation.rb

```diff
@@ -62,6 +62,12 @@ def inject!(digest, data)
           end
 
           result = false
+          # Probably sample here
```

Comment:
Line 57 we try to convert a span to a digest anyways.

Reply:
However we could possibly get the current span with the tracer object?

Reply:
The issue here is that the digest is already created at this point, so even if we run sampling on the trace, the digest won't change.